File: C/Binary_Search.c
#include <stdio.h>

int Binary_Search(int a[], int l, int h, int x){

    if(l > h){ 
        printf("Element does not exist.");
        return 0;
    }

    int mid = (l + h)/2;

    if(x == a[mid]){
        printf("Element exists");
    } else if(x > a[mid]){
        Binary_Search(a, mid+1, h, x);
    } else if(x < a[mid]){
        Binary_Search(a, l, mid-1, x);
    }

    return 0;
}

int main(){

    int a[] = {1, 3, 5, 7, 9};

    Binary_Search(a, 0, 5, 4);

    return 0;
}

File: Python/Binary_Search.py
L = [1,3,5,7,9]

def Binary_Search(List, low, high, x):
    
    if low > high:
        print("Element does not exist.")
        return False
    
    mid = (low+high)//2
    
    if x == List[mid]:
        print("Element exists.")
    elif x > List[mid]:
        Binary_Search(L, mid+1, high, x)
    else:
        Binary_Search(L, low, mid-1, x)
        
Binary_Search(L, 0, 5, 7)
